tests/components/sharkiq/test_config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Test the Shark IQ config flow."""
+import aiohttp
+from sharkiqpy import SharkIqAuthError
+
+from homeassistant import config_entries, setup
+from homeassistant.components.sharkiq.const import DOMAIN
+
+from .const import CONFIG, TEST_PASSWORD, TEST_USERNAME, UNIQUE_ID
+
+from tests.async_mock import MagicMock, PropertyMock, patch
+from tests.common import MockConfigEntry
+
+
+def _create_mocked_ayla(connect=None):
+    """Create a mocked AylaApi object."""
+    mocked_ayla = MagicMock()
+    type(mocked_ayla).sign_in = PropertyMock(side_effect=connect)
+    type(mocked_ayla).async_sign_in = PropertyMock(side_effect=connect)
+    return mocked_ayla
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch("sharkiqpy.AylaApi.async_sign_in", return_value=True), patch(
+        "homeassistant.components.sharkiq.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.sharkiq.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG,
+        )
+
```

Comment:
I think that you need to block till done here too, to make sure that the mocks for async_setup are still active when the entry is set up.
```suggestion
        await hass.async_block_till_done()

```